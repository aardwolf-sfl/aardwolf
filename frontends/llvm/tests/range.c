int range(int *array, int k)
{
    if (k == 0) {
        return 0;
    } else {
        int min = array[0];
        int max = array[0];

        int i;
        for (i = 1; i < k; i++) {
            if (array[i] < min) {
                min = array[i];
            } else if (array[i] > max) {
                max = array[i];
            }
        }

        return max - min;
    }
}
