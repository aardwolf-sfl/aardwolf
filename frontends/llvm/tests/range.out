
function: range

#1 -> #2  ::  defs: %1 / uses:  [@1 1:16-1:16]  { arg }
#2 -> #3  ::  defs: %2 / uses:  [@1 1:27-1:27]  { arg }
#3 -> #4, #5  ::  defs:  / uses: %2 [@1 3:9-3:9]
#4 -> #15  ::  defs: %3 / uses:  [@1 4:9-4:9]
#5 -> #6  ::  defs: %4 / uses: %1[%7] [@1 6:13-6:13]
#6 -> #7  ::  defs: %5 / uses: %1[%7] [@1 7:13-7:13]
#7 -> #8  ::  defs: %6 / uses:  [@1 10:16-10:16]
#8 -> #14, #9  ::  defs:  / uses: %2, %6 [@1 10:9-10:9]
#9 -> #10, #11  ::  defs:  / uses: %1[%6], %4 [@1 11:17-11:17]
#10 -> #13  ::  defs: %4 / uses: %1[%6] [@1 12:21-12:21]
#11 -> #12, #13  ::  defs:  / uses: %1[%6], %5 [@1 13:24-13:24]
#12 -> #13  ::  defs: %5 / uses: %1[%6] [@1 14:21-14:21]
#13 -> #8  ::  defs: %6 / uses: %6 [@1 10:29-10:29]
#14 -> #15  ::  defs: %3 / uses: %4, %5 [@1 18:9-18:9]
#15 ->   ::  defs:  / uses: %3 [@1 20:1-20:1]  { ret }

@1 = tests/range.c
